File: JDash.Mvc.Demo/Scripts/jdash/model/MetadataModel.js
define(['dojo/_base/declare',
		'dojo/_base/lang',
        'dojo/json',
        './ModelBase'
],
	function (declare, lang, Json, ModelBase) {
		var retClass = declare('jdash.model.MetadataModel', [ModelBase], {

			postscript: function (args) {
				args = args || {};

				if (args.created)
					if (lang.isString(args.created))
						args.created = new Date(parseInt(args.created.replace("/Date(", "").replace(")/", ""), 10));
					else
						args.created = new Date(args.created);
				if (args.modified)
					if (lang.isString(args.modified))
						args.modified = new Date(parseInt(args.modified.replace("/Date(", "").replace(")/", ""), 10));
					else
						args.modified = new Date(args.modified);
				if (args.shared)
					if (lang.isString(args.shared))
						args.shared = new Date(parseInt(args.shared.replace("/Date(", "").replace(")/", ""), 10));
					else
						args.shared = new Date(args.shared);


				this.inherited(arguments, [args]);
			},

			serialized: function () {
				var res = this.inherited(arguments);
				lang.mixin(res, {
					group: this.group,
					description: this.description,
					tags: this.tags,
					createdBy: this.createdBy,
					modifiedBy: this.modifiedBy,
					sharedBy: this.sharedBy,
					created: this.created,
					modified: this.modified,
					shared: this.shared,
					userData: this.userData

				});
				return res;
			}
		});


		return retClass;
	});

File: JDash.Core/Models/MetadataModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;

namespace JDash.Models
{
    public class MetadataModel
    {
        public string description { get; set; }
        public string group { get; set; }        
        public List<string> tags { get; set; }
        public string createdBy { get; set; }
        public string modifiedBy { get; set; }
        public string sharedBy { get; set; }
        public DateTime created { get; set; }
        public DateTime? modified { get; set; }
        public DateTime? shared { get; set; }
        public string userData { get; set; }

        public MetadataModel()
        {
            this.created = DateTime.Now;
            //this.createdBy = JDashManager.Provider != null ? JDashManager.Provider.CurrentUser : Thread.CurrentPrincipal.Identity.Name;
            this.createdBy = Thread.CurrentPrincipal.Identity.Name;

            description = "";
            group = "";
            tags = new List<string>();
        }
    }
}
